docs/TECHNICAL_DEBT_REPORT.md

```
@@ -0,0 +1,230 @@
+- Dependency
+- Configuration
+- Code Quality / Architecture
+- AI Agent Compatibility
+- Testing & QA
+- Documentation
+Each item scored (Impact: Low/Med/High, Effort: S/M/L) with qualitative rationale.
+
+## 3. Debt Inventory by Category
+
+### 3.1 Dependency Debt
+| ID | Issue | Detail | Impact | Effort | Notes |
```

markdownlint-cli2 (0.17.2)

Tables should be surrounded by blank lines

(MD058, blanks-around-tables)